coal/mineral.py

```diff
@@ -6,20 +6,25 @@
 # define number of layers in normalized pixel
 NUMBER_OF_LAYERS = 128
 
-# define wavelength range
-WAVELENGTH_MIN = ???
-WAVELENGTH_MAX = ???
+# define wavelength range in nanometers
+WAVELENGTH_MIN = 383.150
+WAVELENGTH_MAX = 2508.200
 
 def normalize(image):
-
     """
-    Return a copy of an MxNxB image (or 1x1xB pixel) with constrained wavelength range and normalized spectra.
+    This takes the file name of the library to be used in the training and returns a
+    classifier
+    Args:
+       image (str): the image to normalize
+    Returns:
+       normalizedImage (LinearTransform): LinearTransform object?
     """
 
     # TODO constrain wavelength range
     # see https://groups.google.com/d/msg/coal-capstone/6oordALy0dA/a_6VIuWbBAAJ
```

Comment:
the AVIRIS header contains an ordered list of wavelengths corresponding to each band/element in each image/pixel/ndarray. if Spy doesn't have a predefined method to constrain the wavelength, we'd need to find the index of the first and last wavelength in our range and [slice](https://docs.scipy.org/doc/numpy/reference/arrays.indexing.html) the array. something like this:

    # get the wavelength list
    # read from image or pass as argument
    wavelengthList = ???

    # get the indices of the minimum and maximum element
    # we can't use list.index() because that searches for an exact match
    i = wavelengthList.indexOfFirstElementGreaterThan(WAVELENGTH_MIN)
    j = wavelengthList.indexOfFirstElementGreaterThan(WAVELENGTH_MAX) - 1

    # constrain wavelength range
    # slice the array from MxNxB to MxNx(j-i) for indices i,j
    image = image[:,:,i:j]

recall that we are doing this because the minimum and maximum wavelengths differ between spectral libraries and AVIRIS images and this could confuse the neural network.

@heidiaclayton let me know if you'd like me to write up some code for this and send you a pull request.